src/map/utils/charutils.cpp

```diff
@@ -766,6 +759,37 @@ namespace charutils
         luautils::OnGameIn(PChar, zoning == 1);
     }
 
+    void LoadSpells(CCharEntity* PChar)
+    {
+        // disable all spells
+        size_t length = sizeof(PChar->m_SpellList.data());
+        memset(PChar->m_SpellList.data(), 0, length);
```

Comment:
okay, I will make a bitset method to do that then instead of memsetting it

On Mon, Oct 12, 2015 at 10:16 PM, bendangelo <notifications@github.com>
wrote:

> In src/map/utils/charutils.cpp
> <https://github.com/DarkstarProject/darkstar/pull/2218#discussion_r41825439>
> :
>
> > @@ -766,6 +759,37 @@ namespace charutils
> >          luautils::OnGameIn(PChar, zoning == 1);
> >      }
> >
> > +    void LoadSpells(CCharEntity* PChar)
> > +    {
> > +        // disable all spells
> > +        size_t length = sizeof(PChar->m_SpellList.data());
> > +        memset(PChar->m_SpellList.data(), 0, length);
>
> Actually I would prefer to keep this just so LoadSpells can be called any
> time and the spells are loaded on a "clean slate"
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DarkstarProject/darkstar/pull/2218/files#r41825439>.
>